robottelo/api/utils.py

```diff
@@ -189,3 +198,256 @@ def one_to_many_names(name):
 
     """
     return set((name, name + '_ids', Inflector().pluralize(name)))
+
+
+def configure_provisioning(org=None, loc=None):
+    """Create and configure org, loc, product, repo, cv, env. Update proxy,
+    domain, subnet, compute resource, provision templates and medium with
+    previously created entities and create a hostgroup using all mentioned
+    entities.
+
+    :param org: Default Organization that should be used in both host
+        discovering and host provisioning procedures
+    :param loc: Default Location that should be used in both host
+        discovering and host provisioning procedures
+    :return: List of created entities that can be re-used further in
+        provisioning or validation procedure (e.g. hostgroup or domain)
+    """
+    # Create new organization and location in case they were not passed
+    if org is None:
+        org = entities.Organization(name=gen_string('alpha')).create()
+        # org_name = org.name
+    if loc is None:
+        loc = entities.Location(
+            name=gen_string('alpha'),
+            organization=[org],
+        ).create()
+        # loc_name = loc.name
+    # Create a new Life-Cycle environment
+    lc_env = entities.LifecycleEnvironment(
+        name=gen_string('alpha'),
+        organization=org
+    ).create()
+    # Create a Product, Repository for custom RHEL6 contents
+    product = entities.Product(
+        name=gen_string('alpha'),
+        organization=org
+    ).create()
+    repo = entities.Repository(
+        name=gen_string('alpha'),
+        product=product,
+        url=settings.rhel7_os
+    ).create()
+
+    # Increased timeout value for repo sync
+    old_task_timeout = entity_mixins.TASK_TIMEOUT
+    entity_mixins.TASK_TIMEOUT = 3600
+    repo.sync()
+
+    # Create, Publish and promote CV
+    content_view = entities.ContentView(
+        name=gen_string('alpha'),
+        organization=org
+    ).create()
+    content_view.repository = [repo]
+    content_view = content_view.update(['repository'])
+    content_view.publish()
+    content_view = content_view.read()
+    promote(content_view.version[0], lc_env.id)
+    entity_mixins.TASK_TIMEOUT = old_task_timeout
+    # Search for puppet environment and associate location
+    environment = entities.Environment(
+        organization=[org.id]).search()[0]
+    environment.location = [loc]
+    environment = environment.update(['location'])
+
+    # Search for SmartProxy, and associate location
+    proxy = entities.SmartProxy().search(
+        query={
+            u'search': u'name={0}'.format(
+                settings.server.hostname)
+        }
+    )[0]
+    proxy.location = [loc]
+    proxy = proxy.update(['location'])
+    proxy.organization = [org]
```

Comment:
I may need to raise PR against nailgun to make some changes as I don't see `org` loc` when I read the created entity. However `read_json` shows all the parameters.
```console
In [11]: environment = Environment(server_config, organization=[org.id]).search()[0]

In [12]: environment
Out[12]: nailgun.entities.Environment(nailgun.config.ServerConfig(url='https://server', verify=False, auth=('admin', 'admin')), id=14, name=u'DB7Rgi')

In [14]: environment.read_json()
Out[14]: 
{u'created_at': u'2016-12-20 09:10:37 UTC',
 u'id': 14,
 u'locations': [{u'description': None,
   u'id': 180,
   u'name': u'PlqyMaHUIc',
   u'title': u'PlqyMaHUIc'}],
 u'name': u'DB7Rgi',
 u'organizations': [{u'description': None,
   u'id': 108,
   u'name': u'wok0jH',
   u'title': u'wok0jH'}],
 u'puppetclasses': [],
 u'template_combinations': [],
 u'updated_at': u'2016-12-20 09:10:37 UTC'}
```
This is not just with one entity but with multiple entities. So can we handle this in another PR ?